packages/profiler-lib/ARCHITECTURE.md

```diff
@@ -0,0 +1,345 @@
+# Profiler Architecture
+
+## Overview
+
+The profiler has been split into two packages with clear separation of concerns:
+
+```
+packages/
+├── profiler-lib/          # Reusable visualization library
+│   ├── src/
+│   │   ├── profiler.ts    # Main API class
+│   │   ├── profile.ts     # Profile data parsing
+│   │   ├── dataflow.ts    # Dataflow graph parsing
+│   │   ├── cytograph.ts   # Cytoscape rendering
+│   │   ├── selection.ts   # Node expansion state
+│   │   ├── metadataSelection.ts # Metric/worker filtering
```

Comment:
ui and data structures to select what information to display about each profile node